setenergy_lsdc.py

```diff
@@ -1002,3 +1008,225 @@ def beam_center_align(transSet='All'):
                 yield from trans_set(transOrgBCU, trans=trans_bcu)
         else:
             yield from trans_set(transOrgBCU, trans=trans_bcu)
+
+def get_fluxKeithley():
+    """
+    Returns Keithley diode current derived flux.
+    """
+    
+    keithFlux = epics.caget('XF:17IDA-OP:FMX{Mono:DCM-dflux}')
+    
+    return keithFlux
+
+
+def set_fluxBeam(flux):
+    """
+    Sets the flux reference field.
+    
+    flux: Beamline flux at sample position for transmisison T = 1.  [ph/s]
+    """
+    
+    error = epics.caput('XF:17IDA-OP:FMX{Mono:DCM-dflux-M}', flux)
+    
+    return error
+
+
+def slit1_flux_reference(flux_df,slit1Gap):
+    """
+    Sets Slit 1 X gap and Slit 1 Y gap to a specified position,
+    and returns flux reference values to a provided pandas DataFrame.
+    
+    Supporting function for fmx_flux_reference()
+    
+    Parameters
+    ----------
+    
+    slit1Gap: float
+        Gap value for Slit 1 X and Y [um]
+    
+    flux_df: pandas DataFrame with fields:
+        Slit 1 X gap [um]
+        Slit 1 Y gap [um]
+        Keithley current [A]
+        Keithley flux [ph/s]
+        BPM1 sum [A]
+        BPM4 sum [A]
+        
+    """
+    yield from bps.mv(slits1.x_gap, slit1Gap, slits1.y_gap, slit1Gap, wait=True)
+    time.sleep(2.0) # wait so the slits are definitely done moving and the Keithley reading is stable
+    
+    flux_df.at[slit1Gap, 'Slit 1 X gap [um]'] = slit1Gap
+    flux_df.at[slit1Gap, 'Slit 1 Y gap [um]'] = slit1Gap
+    flux_df.at[slit1Gap, 'Keithley current [A]'] = keithley.get()
+    flux_df.at[slit1Gap, 'Keithley flux [ph/s]'] = get_fluxKeithley()
+    flux_df.at[slit1Gap, 'BPM1 sum [A]'] = bpm1.sum_all.get()
+    # TEMP FIX: flux_df.at[slit1Gap, 'BPM4 sum [A]'] = bpm4.sum_all.get()
+    flux_df.at[slit1Gap, 'BPM4 sum [A]'] = 0
+
+def fmx_flux_reference(slit1GapList = [2000, 1000, 600, 400], slit1GapDefault = 1000, transSet='All'):
+    """
+    Sets Slit 1 X gap and Slit 1 Y gap to a list of settings,
+    and returns flux reference values in a pandas DataFrame.
```

Comment:
this line suggests that the values are returned to the user/caller - but it's actually just storing the global for future use.

Reply:
Fixed in https://github.com/NSLS-II/lsdc/pull/300/commits/323d1c837e0a6805b249d5d587d4194e90727675